### PR TITLE
[SDK-1699] Fix ID token validation for auth_time

### LINKED ISSUE #489: auth_time and max_age are provided in seconds
https://github.com/auth0/auth0-spa-js/blob/e6c3ba188d55576bed4ecf34e7c82d819253f0df/src/jwt.ts#L167

here is the issue, where `auth_time` and `max_age` as per docs should be a seconds
and division by 1000 breaking token validation

### PR DESCRIPTION
**Pull request by stevehobbsdev**
### Description

The `auth_time` claim was being incorrectly validated in milliseconds, where this should be seconds.

### References

Fixes #489 

### Testing

The relevant test was also incorrectly suppling test data in ms, so this has been updated. The test also only validated half of the validation message (the part which doesn't include the time), so this has been improved to correctly validate the entire string.

- [X] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `master`

### CODE REVIEW

File: __tests__/jwt.test.ts [hunk 1]
```diff
@@ -274,19 +289,32 @@ describe('jwt', () => {
...
     const id_token = await createJWT({ ...DEFAULT_PAYLOAD });
+
     expect(() => verify({ ...verifyOptions, id_token, max_age: 123 })).toThrow(
       'Authentication Time (auth_time) claim must be a number present in the ID token when Max Age (max_age) is specified'
     );
   });
+
   it('validate auth_time + max_age is in the future', async () => {
-    const yesterday = new Date();
+    const yesterday = new Date(now);
```
[the remark this answers is not in the document]
Comment (stevehobbsdev) [reply __tests__/jwt.test.ts:299]: Hmm, I did that to pin the date to the exact one I specified at the top of the test. Looking at it, I'll just refactor this to `Date.now()`.

File: __tests__/jwt.test.ts [hunk 2]
```diff
@@ -274,19 +289,32 @@ describe('jwt', () => {
...
-      auth_time: yesterday.getTime()
+      auth_time: authTime
     });
-    expect(() => verify({ ...verifyOptions, id_token, max_age: 1 })).toThrow(
-      'Authentication Time (auth_time) claim in the ID token indicates that too much time has passed since the last end-user authentication.'
+
+    expect(() =>
+      verify({ ...verifyOptions, id_token, max_age: maxAge, leeway })
+    ).toThrow(
+      `Authentication Time (auth_time) claim in the ID token indicates that too much time has passed since the last end-user authentication. Currrent time (${new Date(
```
[the remark this answers is not in the document]
Comment (stevehobbsdev) [reply __tests__/jwt.test.ts:315]: Actually I meant to fix that, but you're right, people could be relying on that message string - what do you suggest?

[the remark this answers is not in the document]
Comment (adamjmcgrath) [reply __tests__/jwt.test.ts:315]: Let's leave it for now

File: __tests__/jwt.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamjmcgrath) [reply __tests__/jwt.test.ts:299]: Yeah, you can trigger the error with a date of anything older than 61 seconds before now. It's confusing when you pin it to exactly 24hrs before your mocked `now`

**Comment by MrPrashantT** [comment]
Hi team, any indication on when this fix will be incorporated into a new release?

**Comment by stevehobbsdev** [comment]
Hi @MrPrashantT, this is now available in the [v1.10.0 release](https://github.com/auth0/auth0-spa-js/releases/tag/v1.10.0) that was released yesterday.

This was released to NPM yesterday but I've just tagged it now - thanks for the reminder!

**Pull request merged**